mypy/dmypy_server.py

```diff
@@ -200,8 +200,8 @@ def serve(self) -> None:
                 json.dump({'pid': os.getpid(), 'connection_name': server.connection_name}, f)
                 f.write('\n')  # I like my JSON with a trailing newline
             while True:
-                with server:
-                    data = receive(server)
+                with server as connection:
```

Comment:
These semantics of `__enter__` are a little too magical. I'd prefer to see something like `with server.accept_connection() as connection` here.